lib/features/manage_account/data/datasource_impl/notification_datasource_impl.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:permission_handler/permission_handler.dart';
+import 'package:tmail_ui_user/features/manage_account/data/datasource/notification_datasource.dart';
+import 'package:tmail_ui_user/features/manage_account/data/local/notification_setting_cache_manager.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/exceptions/notification_exceptions.dart';
+import 'package:tmail_ui_user/main/exceptions/exception_thrower.dart';
+import 'package:tmail_ui_user/main/permissions/permission_service.dart';
+
+class NotificationDataSourceImpl implements NotificationDataSource {
+  final NotificationSettingCacheManager _notificationSettingCacheManager;
+  final PermissionService _permissionService;
+  final ExceptionThrower _exceptionThrower;
+
+  NotificationDataSourceImpl(
+    this._notificationSettingCacheManager,
+    this._permissionService,
+    this._exceptionThrower);
+
+  @override
+  Future<bool> getAppNotificationSetting() {
+    return Future.sync(() async {
+      final cache = _notificationSettingCacheManager.getAppNotificationSettingCache();
+      if (cache != null) return cache;
+
+      final notificationPermissionGranted = await _permissionService
+        .isGranted(Permission.notification);
+      await _notificationSettingCacheManager
+        .persistAppNotificationSettingCache(notificationPermissionGranted);
+      return notificationPermissionGranted;
+    }).catchError(_exceptionThrower.throwException);
+  }
+  
+  @override
+  Future<void> toggleAppNotificationSetting(bool isEnabled) async {
+    return Future.sync(() async {
+      if (!isEnabled) {
+        return await _notificationSettingCacheManager.persistAppNotificationSettingCache(false);
+      }
+
+      final notificationPermissionGranted = await _permissionService
+        .isGranted(Permission.notification);
+      if (notificationPermissionGranted) {
+        await _notificationSettingCacheManager
+          .persistAppNotificationSettingCache(true);
+      } else {
+        throw const NotificationPermissionDeniedException();
+      }
+    }).catchError(_exceptionThrower.throwException);
+  }
+  
+  @override
+  Future<void> attemptToggleSystemNotificationSetting() async {
+    return Future.sync(() async {
+      await openAppSettings();
```

Comment:
Can we go to more specific system settings for notification as gmail?

![IMG_4764](https://github.com/linagora/tmail-flutter/assets/6462404/82fb48fe-1ff1-401e-a59f-4ca3560bc155)

Reply:
Can. Will update demo.

Reply:
Updated